src/stp/plays/defensive/DefendShot.cpp

```diff
@@ -23,52 +23,118 @@ DefendShot::DefendShot() : Play() {
+        std::make_unique<role::Harasser>(role::Harasser("harasser")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_1")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_2")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_3")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_4")),
     };
 }
 
 uint8_t DefendShot::score(PlayEvaluator &playEvaluator) noexcept {
     auto world = playEvaluator.getWorld();
     auto field = world->getField().value();
     auto enemyRobot = world->getWorld()->getRobotClosestToBall(world::them);
+    auto goalVisibility =
+        FieldComputations::getPercentageOfGoalVisibleFromPoint(field, true, enemyRobot->get()->getPos(), world->getWorld().value(), enemyRobot->get()->getId(), false);
     auto position = distanceFromPointToLine(field.getBottomLeftCorner(), field.getTopLeftCorner(), enemyRobot->get()->getPos());
-    return 255 * (field.getFieldLength() - position) / field.getFieldLength();
+    return 255 * (field.getFieldLength() - position) / field.getFieldLength() * goalVisibility/100;
 }
 
 Dealer::FlagMap DefendShot::decideRoleFlags() const noexcept {
     Dealer::FlagMap flagMap;
 
     Dealer::DealerFlag closestToBallFlag(DealerFlagTitle::CLOSEST_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
     Dealer::DealerFlag closeToOurGoalFlag(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::HIGH_PRIORITY);
+    Dealer::DealerFlag notImportant(DealerFlagTitle::NOT_IMPORTANT, DealerFlagPriority::LOW_PRIORITY);
 
     flagMap.insert({"keeper", {DealerFlagPriority::KEEPER, {}}});
-    flagMap.insert({"harassing_defender", {DealerFlagPriority::HIGH_PRIORITY, {closestToBallFlag}}});
+    flagMap.insert({"harasser", {DealerFlagPriority::HIGH_PRIORITY, {closestToBallFlag}}});
+    flagMap.insert({"waller_1", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"waller_2", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"waller_3", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
     flagMap.insert({"defender_1", {DealerFlagPriority::MEDIUM_PRIORITY, {closeToOurGoalFlag}}});
     flagMap.insert({"defender_2", {DealerFlagPriority::MEDIUM_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"midfielder_1", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_2", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_3", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_4", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
 
     return flagMap;
 }
 
 void DefendShot::calculateInfoForRoles() noexcept {
+    calculateInfoForWallers();
     calculateInfoForDefenders();
     calculateInfoForHarassers();
     calculateInfoForKeeper();
 }
 
+void DefendShot::calculateInfoForWallers() noexcept{
```

Comment:
Do you think using the determineWallPosition function in PositionComputations could be used here? I'm not exactly sure how the results would differ, but it might be interesting to look into